lib/caffeinate/dripper/periodical.rb

```diff
@@ -11,12 +11,23 @@ module ClassMethods
         def periodical(action_name, every:, start: -> { ::Caffeinate.config.time_now }, **options, &block)
           options[:start] = start
           options[:every] = every
-          drip(action_name, options, &block)
+          periodical_drip(action_name, **options, &block)
+
           after_send do |mailing, _message|
-            if mailing.drip.action == action_name
-              next_mailing = mailing.dup
-              next_mailing.send_at = mailing.drip.send_at(mailing)
-              next_mailing.save!
+            if (mailing.drip.action == action_name) && (thing = mailing.drip.options[:if])
+              if OptionEvaluator.new(thing, mailing.drip, mailing).call
+                if mailing.drip.action == action_name
+                  next_mailing = mailing.dup
+                  next_mailing.send_at = mailing.drip.send_at(mailing)
+                  next_mailing.save!
+                end
+              end
+            else
+              if mailing.drip.action == action_name
+                next_mailing = mailing.dup
+                next_mailing.send_at = mailing.drip.send_at(mailing)
+                next_mailing.save!
+              end
```

Comment:
```suggestion
            if (mailing.drip.action == action_name)
              return unless (thing = mailing.drip.options[:if]) && OptionEvaluator.new(thing, mailing.drip, mailing).call
              next_mailing = mailing.dup
              next_mailing.send_at = mailing.drip.send_at(mailing)
              next_mailing.save!
            end
```

Reply:
`if` isn't required and we still want to queue things up, no?

Reply:
returns break proc

Reply:
Ah; I always forget about that. `next` works for those situations, but having the more verbose code is also fine 😛 either way!